### ISSUE TITLE
[Debugger] Attaching with GDB doesn't work

### ISSUE THREAD

**Issue by Tim---** [NationalSecurityAgency/ghidra#2735]
**Describe the bug**

Attaching to a process doesn't work with GDB because of a confusion in the PID format.

I narrowed down the problem here:

https://github.com/NationalSecurityAgency/ghidra/blob/debugger/Ghidra/Debug/Debugger/src/main/java/ghidra/app/plugin/core/debug/gui/objects/components/DebuggerAttachDialog.java#L137

If I list available processes and try to attach to PID 123, `proc.getName()` will contain `[123]`. However, `getPid` assumes the PID is hexadecimal, and converts it to decimal. Ghidra will then ask GDB to attach to PID 291 (0x123), which fails.

I don't know if that's an error, or if it depends on versions of GDB. On my GDB, `-list-thread-groups --available` returns `id` in decimal, but maybe that's not always the case ?

**To Reproduce**
1. Start Ghidra with the debugger
2. List the "Available" processes
3. Select a process with eg. pid 123 (which is the same as the PID given by `ps`).
4. Attach to the process
5. Attaching fails: an inferior is created but has no content.

**Expected behavior**
Attach to the correct process :).

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.2 LTS
 - Java Version: openjdk version "11.0.9.1" 2020-11-04
 - Ghidra Version: debugger branch
 - GDB Version: GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2

**Comment by d-millar** [comment]
Thanks - will put a ticket in for this.  In the meantime, you can open the Connectors node in the Objects Provider, select the one for Attach and put the pid in manually.

**Comment by Tim---** [comment]
Thanks for your response.
However, I don't see a Connectors node for GDB. Is it for windbg only, or did I mess my build really bad ?
In any case I can work around by launching the binary directly. I just stumbled upon this bug while testing the debugger features.

**Comment by d-millar** [comment]
Oh - ha ha ha - you're right.  That doesn't exist for gdb - apologies.  OK, bumping up the priority of fixing this!

**Comment by class101** [comment]
Thank you @Tim--- always super handy to share solutions, it helped me a lot to pinpoint this issue.

If by the way, anybody on Arch Linux is reading this, I have set up an AUR package for the branch debugger here

https://aur.archlinux.org/packages/ghidra-dev

It automatically updates and packages to each newer commits and includes 2 patches, the one you mentionned @Tim--- and another less critical one with the schema validator reporting uninitialized attributes.

I also made sure to offer an easy way to run Ghidra as root with a second desktop shortcut and a newer dependency to polkit as attaching to process works best with highest privileges.

Thank you @d-millar and others for all the work on the debugger, it works very well and is very handy. I'm learning slowly your project architecture so I hope I will be able to help you soon.
